scalableminds/webknossos
# add 'boundingBox' to NmlParser and NmlWriter

### Mailable description of changes:
 - We now write the bounding box of a SkeletonTracing into the nml.
   Example: `<taskBoundingBox topLeftX="0" topLeftY="0" topLeftZ="0" width="512" height="512" depth="512" />`
 - We also parse the bounding box when uploading an nml file.
   - There are four possible scenarios when uploading a nml file because it is still possible to specify the bounding box explicitly in the form-value `BoundinBox`:
   1. the bounding box is specified in the form-value but NOT in the parsed nml 
   => the bounding box of the form-value is used
   2.  the bounding box is specified in the form-value AND in the parsed nml 
   => the bounding box of the form-value is used
   3.  the bounding box is NOT specified in the form-value but in the parsed nml 
   => the bounding box of the parsed NML is used
   4.  the bounding box is NEITHER specified in the form-value nor in the parsed nml 
   => no bounding box is used

### Steps to test:
- Export the Nml of a Task via `/api/annotations/:typ/:id/download ` 
   - make sure the nml has a field `boundingBox` if the task has a boundingBox (this is optional)
- Import the Task again (Administration >> Tasks >> add Task >> 'Upload nml file')
   - try setting the boundingBox in the form explicit (the form value has a higher priority)
   - make sure the new Task has the specified boundingBox (from the form or the nml)

### Issues:
- fixes #2802 

------
